packages/calcite-components/src/demos/_assets/locales.ts

```diff
@@ -1,236 +1,289 @@
-import { NumberingSystem } from "../../utils/locale";
+import { NumberingSystem, SupportedLocale } from "../../utils/locale";
+import { HourFormat } from "../../utils/time";
 
 interface Locale {
   name: string;
-  locale: string;
+  locale: SupportedLocale;
   dir?: "ltr" | "rtl";
   numberingSystem?: NumberingSystem;
+  hourFormat: HourFormat;
```

Comment:
By reference do you mean installing something like `cldr-json` instead of defining the data ourselves for each locale?

Reply:
[When I look at the Arabic entry in `cldr-json`](https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-dates-full/main/ar/ca-generic.json), for example, there's actually no clear data field that says that this locale prefers the 12-hour clock, so I'm not sure that referencing the CLDR for our `HourFormat` interface is entirely necessary.  Because CLDR specifies all the possible ways of formatting the time value (both with and without meridiems), there's no real way to determine the preferred format, other than gleaning it from one of the defined time formats in the file, and even in that case there's no way to know if that's the locale's default/preferred one.

Reply:
I _think_ that info is covered [here](https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/timeData.json) (see https://cldr.unicode.org/translation/date-time/date-time-patterns#basic-time-formats). Could we reference it similarly to other comments you've added? cc @annierm18

Reply:
Cool!  I never discovered that area of the CLDR, good find!  Looks like a good place to link to, and I'll add explanation text that could help future people looking at this code to say that `h` means 12-hour and `H` refers to 24-hour.  It's weird to see the lang object keys in all caps, but there's probably a good reason for that...